python/pyjava/data/datasource.py

```diff
@@ -16,17 +17,36 @@ def __init__(self, data_refs: List[str], num_tables_per_block: int = 1):
         self.data_refs = data_refs
         self.block_refs = []
 
+        def merged_pa_generator(pa_generator):
+            merged_tables = []
+            while True:
+                try:
+                    for i in range(0, 7):
```

Comment:
The number of tables merged should be configured by parameter `num_tables_per_block`. The value is 1 by default.